### PR TITLE
Various tidy ups and deinit path improvements.

### PR DESCRIPTION
**Pull request by glennw**
* Change delete_program to move the program.
* Remove hashmap for VAOs.
* Add deinit path for VAOs similar to programs.
* Update renderer and debug render code.
* Remove shader watch code (can re-add later if we get it working again).

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1582)

### CODE REVIEW

File: webrender/src/device.rs [hunk 1]
```diff
@@ -366,32 +364,18 @@ impl Drop for Program {
     }
 }
 
-struct VAO {
-    gl: Rc<gl::Gl>,
```
Comment (kvark) [inline webrender/src/device.rs]: niiice

File: webrender/src/device.rs [hunk 2]
```diff
@@ -366,32 +364,18 @@ impl Drop for Program {
...
-            // todo(gw): maybe make these their own type with hashmap?
-            self.gl.delete_buffers(&[self.ibo_id.0]);
-        }
-        if self.owns_vertices {
-            self.gl.delete_buffers(&[self.main_vbo_id.0]);
-        }
-        if self.owns_instances {
-            self.gl.delete_buffers(&[self.instance_vbo_id.0])
-        }
+        debug_assert!(thread::panicking() || self.id == 0);
```
Comment (kvark) [inline webrender/src/device.rs]: need to add a message saying that `deinit` was not called
not a blocked, can do as follow up

**Comment by glennw** [comment]
Rebased and updated the deinit assert comment.

@bors-servo r=kvark

**Pull request merged**